src/stk/molecular/molecules/building_block.py

```diff
@@ -553,14 +552,26 @@ def _init_from_rdkit_mol(
             Atom(a.GetIdx(), a.GetAtomicNum(), a.GetFormalCharge())
             for a in molecule.GetAtoms()
         )
-        bonds = tuple(
-            Bond(
-                atom1=atoms[b.GetBeginAtomIdx()],
-                atom2=atoms[b.GetEndAtomIdx()],
+
+        bonds = []
+        for b in molecule.GetBonds():
+            # Check atom ordering.
+            # Atom2 in stk should be the metal atom.
```

Comment:
atom2 not Atom2 I think - since its referring to a variable called atom2, also the comment should explain *why* it should be the metal atom.